src/vm/errors/vm_errors.rs

```diff
@@ -93,6 +93,8 @@ pub enum VirtualMachineError {
     NoImm,
     #[error("Execution reached the end of the program. Requested remaining steps: {0}.")]
     EndOfProgram(usize),
+    #[error("Couldnt reached the end of the program. Executed steps: {0}.")]
+    StepsLimit(u64),
```

Comment:
are we sure that u64 is enough?

Reply:
Couldnt -> Could not (formal english please)

Reply:
I think u64 will be enough, u64::MAX = 18_446_744_073_709_551_615u64
Also `current_step ` and the other methods uses `usize` for the step counters